src/main/java/com/jadventure/game/repository/NpcRepository.java

```diff
@@ -0,0 +1,123 @@
+            String id = entry.getKey();
+            JsonObject npcData = entry.getValue().getAsJsonObject();
+            String name = npcData.get("name").getAsString();
+            int health =  npcData.get("health").getAsInt();
+            int damage = npcData.get("damage").getAsInt();
+            int armour = npcData.get("armour").getAsInt();
+            int healthMax = npcData.get("healthMax").getAsInt();
+            int level = npcData.get("level").getAsInt();
+            int intelligence = npcData.get("intelligence").getAsInt();
+            int dexterity = npcData.get("dexterity").getAsInt();
+            int stealth = npcData.get("stealth").getAsInt();
+            int strength = npcData.get("strength").getAsInt();
+            int gold = 0;
+            if (npcData.has("gold")) {
+                gold = npcData.get("gold").getAsInt();
+            }
+            // construct npc
+            NPC npc = new NPC();
+            npc.setId(id);
+            npc.setName(name);
+            npc.setHealth(health);
+            npc.setDamage(damage);
+            npc.setArmour(armour);
+            npc.setHealthMax(healthMax);
+            npc.setLevel(level);
+            npc.setIntelligence(intelligence);
+            npc.setDexterity(dexterity);
+            npc.setStealth(stealth);
+            npc.setStrength(strength);
+            npc.setGold(gold);
+            float maxWeight = (float)Math.sqrt(strength*300);
+            npc.setStorage( new Storage(maxWeight) );
+            if (npcData.has("sellLimit") && npcData.has("items")) {
+                int itemLimit = npcData.get("sellLimit").getAsInt();
+                int i = 0;
+                npc.setItems(npcData, itemLimit, i);
+            }
+            JsonArray alliesJson = npcData.get("allies").getAsJsonArray();
+            List<String> allies = new ArrayList<>();
+            for (JsonElement ally : alliesJson) {
+                allies.add(ally.getAsString());
+            }
+            npc.setAllies(allies);
+            List<String> enemies = new ArrayList<>();
+            JsonArray enemiesJson = npcData.get("enemies").getAsJsonArray();
+            for (JsonElement enemy : enemiesJson) {
+                enemies.add(enemy.getAsString());
+            }
+            npc.setEnemies(enemies);
+            npcMap.put(id,npc);
+        }
+    }
+    
+    private static NpcRepository npcRepository = null;
+    public static NpcRepository createRepo() {
+        if ( npcRepository == null) {
+            File dataFile = new File(fileName);
+            if (! dataFile.exists()) {
+            	throw new RuntimeException("File '" + dataFile + "' does not exist.");
+            }
+
+            npcRepository = new NpcRepository();
+            npcRepository.load(dataFile);
+        }
+        return npcRepository;
+    }
+    
+    public NPC getNpc(String npcId) {
+        return npcMap.get(npcId);
```

Comment:
In following with the template set out in ItemRepository, can this rather throw an RepositoryException than return null if the NPC is not found? Only return null if the parameter is null or blank